persistent-sqlite/Database/Sqlite.hs

```diff
@@ -183,24 +183,28 @@ sqlError maybeConnection functionName error = do
     , seDetails = details
     }
 
-foreign import ccall "sqlite3_open"
-  openC :: CString -> Ptr (Ptr ()) -> IO Int
-openError :: Text -> IO (Either Connection Error)
-openError path' = do
+foreign import ccall "sqlite3_open_v2"
+  openC :: CString -> Ptr (Ptr ()) -> Int -> CString -> IO Int
+openError :: Text -> Bool -> IO (Either Connection Error)
+openError path' readOnlyFlag = do
+  -- https://www.sqlite.org/c3ref/open.html
+  -- 1 = SQLITE_OPEN_READONLY
+  -- 6 = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE (behavior of sqlite3_open)
+  let flag = if readOnlyFlag then 1 else 6
```

Comment:
Doesn't seem ideal to hardcode bitmask combinations. What about listing out the potential values and bitwise-ORing them manually? This approach would also make it easier to support the rest of the SQLite bitmask options, too. Something like this maybe?

```haskell
import Data.Bits

sqliteFlagReadWrite = 2
sqliteFlagCreate = 4

2 .|. 4 -- 6
```